cmd/elastic_ip_list.go

```diff
@@ -50,51 +49,47 @@ func (c *elasticIPListCmd) cmdPreRun(cmd *cobra.Command, args []string) error {
 }
 
 func (c *elasticIPListCmd) cmdRun(_ *cobra.Command, _ []string) error {
-	var zones []string
+	client := globalstate.EgoscaleV3Client
+	ctx := gContext
+
+	resp, err := client.ListZones(ctx)
+	if err != nil {
+		return err
+	}
+	zones := resp.Zones
 
 	if c.Zone != "" {
-		zones = []string{c.Zone}
-	} else {
-		zones = utils.AllZones
+		endpoint, err := client.GetZoneAPIEndpoint(ctx, c.Zone)
+		if err != nil {
+			return err
+		}
+
+		zones = []v3.Zone{{APIEndpoint: endpoint}}
 	}
 
-	out := make(elasticIPListOutput, 0)
-	res := make(chan elasticIPListItemOutput)
-	done := make(chan struct{})
+	output := make(elasticIPListOutput, 0)
 
-	go func() {
-		for nlb := range res {
-			out = append(out, nlb)
-		}
-		done <- struct{}{}
-	}()
-	err := utils.ForEachZone(zones, func(zone string) error {
-		ctx := exoapi.WithEndpoint(gContext, exoapi.NewReqEndpoint(account.CurrentAccount.Environment, zone))
+	for _, zone := range zones {
+		c := client.WithEndpoint(zone.APIEndpoint)
 
-		list, err := globalstate.EgoscaleClient.ListElasticIPs(ctx, zone)
+		resp, err := c.ListElasticIPS(ctx)
 		if err != nil {
-			return fmt.Errorf("unable to list Elastic IP addresses in zone %s: %w", zone, err)
+			_, _ = fmt.Fprintf(os.Stderr,
+				"warning: errors during listing, results might be incomplete.\n%s\n", err) // nolint:golint
+			continue
 		}
 
-		for _, e := range list {
-			res <- elasticIPListItemOutput{
-				ID:        *e.ID,
-				IPAddress: e.IPAddress.String(),
-				Zone:      zone,
-			}
+		for _, elasticIP := range resp.ElasticIPS {
+			output = append(output, elasticIPListItemOutput{
+				ID:          elasticIP.ID,
+				IPAddress:   elasticIP.IP,
+				Zone:        string(zone.Name),
```

Comment:
no need the `string()` if you updated the type in output, this such change apply to this PR and other one :)